File: 189-旋转数组/189-2.cpp
class Solution {
public:
	
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        if(nums.empty() || (k %=n) == 0){
            return;
        }
        for(int i = 0;i<n-k;i++){
            nums.push_back(nums[0]);
            nums.erase(nums.begin());
        }
    }
};

File: 189-旋转数组/189-2.java
class Solution {
  public void rotate(int[] nums, int k) {
      int n = nums.length;
      k %= n;
      reverse(nums, n-k,n-1);
      reverse(nums, 0,n-k-1);
      reverse(nums,0,n-1);
  }
  void reverse(int[]nums, int l, int r){
      while(l<r){
          int tmp = nums[l];
          nums[l] = nums[r];
          nums[r] = tmp;
          l++;
          r--;
      }
  }
}